packages/pubsub/src/topic.js

```diff
@@ -348,218 +378,43 @@ Topic.prototype.getSubscriptions = function(options, callback) {
- * @param {number} options.ackDeadlineSeconds - The maximum time after
- *     receiving a message that you must ack a message before it is redelivered.
- * @param {boolean=} options.autoAck - Automatically acknowledge the message
- *     once it's pulled. (default: false)
- * @param {string} options.encoding - When pulling for messages, this type is
- *     used when converting a message's data to a string. (default: 'utf-8')
- * @param {number} options.interval - Interval in milliseconds to check for new
- *     messages. (default: 10)
- * @param {number} options.maxInProgress - Maximum messages to consume
- *     simultaneously.
- * @param {number|date} options.messageRetentionDuration - Set this to override
- *     the default duration of 7 days. This value is expected in seconds.
- *     Acceptable values are in the range of 10 minutes and 7 days.
- * @param {string} options.pushEndpoint - A URL to a custom endpoint that
- *     messages should be pushed to.
- * @param {boolean} options.retainAckedMessages - If set, acked messages are
- *     retained in the subscription's backlog for 7 days (unless overriden by
- *     `options.messageRetentionDuration`). Default: `false`
- * @param {number} options.timeout - Set a maximum amount of time in
- *     milliseconds on an HTTP request to pull new messages to wait for a
- *     response before the connection is broken.
- * @param {function} callback - The callback function.
- *
- * @example
- * // Without specifying any options.
- * topic.subscribe('newMessages', function(err, subscription, apiResponse) {});
- *
- * //-
- * // Omit the name to have one generated automatically. All generated names
- * // share a common prefix, `autogenerated-`.
- * //-
- * topic.subscribe(function(err, subscription, apiResponse) {
- *   // subscription.name = The generated name.
- * });
- *
- * // With options.
- * topic.subscribe('newMessages', {
- *   ackDeadlineSeconds: 90,
- *   autoAck: true,
- *   interval: 30
- * }, function(err, subscription, apiResponse) {});
- *
- * //-
- * // If the callback is omitted, we'll return a Promise.
- * //-
- * topic.subscribe('newMessages').then(function(data) {
- *   var subscription = data[0];
- *   var apiResponse = data[1];
  * });
  */
-Topic.prototype.subscribe = function(subName, options, callback) {
-  this.parent.subscribe(this, subName, options, callback);
+Topic.prototype.publisher = function(options) {
+  return new Publisher(this, options);
 };
 
 /**
  * Create a Subscription object. This command by itself will not run any API
  * requests. You will receive a {module:pubsub/subscription} object,
  * which will allow you to interact with a subscription.
  *
+ * @throws {Error} If subscription name is omitted.
+ *
  * @param {string} name - Name of the subscription.
  * @param {object=} options - Configuration object.
- * @param {boolean=} options.autoAck - Automatically acknowledge the message
```

Comment:
We need to show the options here for `flowControl` (and whatever else is relevant).